client/src/rpc_client.rs

```diff
@@ -25,12 +26,187 @@ use solana_sdk::{
     signature::{KeypairUtil, Signature},
     transaction::{self, Transaction, TransactionError},
 };
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::mpsc::{channel, Receiver};
+use std::sync::{Arc, RwLock};
+use std::thread::JoinHandle;
 use std::{
     error, io,
     net::SocketAddr,
     thread::sleep,
     time::{Duration, Instant},
 };
+use thiserror::Error;
+use tungstenite::{client::AutoStream, connect, Message, WebSocket};
+use url::{ParseError, Url};
+
+#[derive(Debug, Error)]
+pub enum PubsubClientError {
```

Comment:
All this new code in rpc_client seems complex enough to justify a separate module: pubsub_client.rs, perhaps?
Any reason it needs to stay here?